kolibri/core/content/api.py

```diff
@@ -1321,6 +1326,82 @@ def consolidate(self, items, queryset):
         return sorted_items
 
 
+class ContentRequestViewset(ReadOnlyValuesViewset, CreateModelMixin):
+    serializer_class = serializers.ContentDownloadRequestSeralizer
+
+    pagination_class = OptionalPageNumberPagination
+
+    values = (
+        "id",
+        "requested_at",
+        "reason",
+        "contentnode_id",
+        "metadata",
+        "status",
+        "facility",
+        "source_id",
+    )
+
+    def get_queryset(self):
+        return ContentDownloadRequest.objects.filter(source_id=self.request.user.id)
+
+    def annotate_queryset(self, queryset):
+        return (
+            queryset.annotate(
+                has_removal=Exists(
+                    ContentRemovalRequest.objects.filter(
+                        source_model=OuterRef("source_model"),
+                        source_id=OuterRef("source_id"),
+                        contentnode_id=OuterRef("contentnode_id"),
+                        requested_at__gte=OuterRef("requested_at"),
+                    )
```

Comment:
So the change would be here instead:

```suggestion
                    ).exclude(status=ContentRequestStatus.Failed)
```